target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-mx5300.dts

```diff
@@ -0,0 +1,539 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include "ipq8074-ess.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Linksys MX5300";
+	compatible = "linksys,mx5300", "qcom,ipq8074";
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		/*
+		 * Aliases as required by u-boot
+		 * to patch MAC addresses
+		 */
+		ethernet0 = &dp1;
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5;
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue, &led_system_red;
+		led-failsafe = &led_system_red;
+		led-upgrade = &led_system_green;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " root=/dev/ubiblock0_0 rootfstype=squashfs ro";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset-button {
+			label = "reset";
+			gpios = <&tlmm 67 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps-button {
+			label = "wps";
+			gpios = <&tlmm 54 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&tlmm {
+	button_pins: button-state {
+		pins = "gpio54", "gpio67";
+		function = "gpio";
+		drive-strength = <8>;
+		bias-pull-up;
+	};
+
+	mdio_pins: mdio-state {
+		mdc-pins {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio-pins {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	spi_3_pins: spi-3-state {
```

Comment:
Are you sure that this gets properly applied?

Because, this is completely against the bindings the way its currently structured.

Reply:
I'm not sure how pins should be described. Here is the OEM dts:
- https://gist.github.com/testuser7/3db5a8ff8b621ae1a216dd919ce2113a#file-mx5300-dts-L3134-L3155
- https://gist.github.com/testuser7/3db5a8ff8b621ae1a216dd919ce2113a#file-mx5300-dts-L2075-L2106

Reply:
I dont care about vendor DTS-es, they are usually broken in so many ways.

They should be like:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/arch/arm64/boot/dts/qcom/ipq8074.dtsi?h=next-20231128#n343

Just change the naming and pins to the correct ones for this QUP.

Those Quartz pins should be a separate node that you can just include along spi3_pins in the spidev node

Reply:
It's not obvious to me how the pins should be configured.
Is this correct?
```
	spi_3_default: spi-3-default {
		mux {
			pins = "gpio50", "gpio52", "gpio53";
			function = "blsp3_spi";
			drive-strength = <8>;
			bias-disable;
		}

		spi_cs {
			pins = "gpio22";
			function = "blsp3_spi2";
			drive-strength = <8>;
			bias-disable;
		};
	};

	spi_3_interrupt: spi-3-interrupt {
		pins = "gpio48";
		function = "gpio";
		bias-disable;
		input;
	};

	spi_3_reset: spi-3-reset {
		pins = "gpio21";
		function = "gpio";
		bias-disable;
		output-low;
	};
   
&blsp1_spi4 {
	pinctrl-names = "default", "interrupt", "reset";
	pinctrl-0 = <&spi_3_default>;
	pinctrl-1 = <&spi_3_interrupt>;
	pinctrl-2 = <&spi_3_reset>;
	...
```
or maybe this way?
```
	spi_3_pins: spi-3-state {
		mux {
			pins = "gpio50", "gpio52", "gpio53";
			function = "blsp3_spi";
			drive-strength = <8>;
			bias-disable;
		}

		quartz_interrupt {
			pins = "gpio48";
			function = "gpio";
			bias-disable;
			input;
		};

		quartz_reset {
			pins = "gpio21";
			function = "gpio";
			bias-disable;
			output-low;
		};

		spi_cs {
			pins = "gpio22";
			function = "blsp3_spi2";
			drive-strength = <8>;
			bias-disable;
		};
	};
	
&blsp1_spi4 {
	pinctrl-names = "default";
	pinctrl-0 = <&spi_3_pins>;
	...
```

Reply:
```
&tlmm {
	spi_3_pins: spi-3-state {
		spi-pins {
			pins = "gpio50", "gpio52", "gpio53";
			function = "blsp3_spi";
			drive-strength = <8>;
			bias-disable;
		};

		cs-pins {
			pins = "gpio22";
			function = "blsp3_spi2";
			drive-strength = <8>;
			bias-disable;
		};
	};

	quartz_pins: quartz-state {
		interrupt-pins {
			pins = "gpio48";
			function = "gpio";
			bias-disable;
			input;

		};

		reset-pins {
			pins = "gpio21";
			function = "gpio";
			bias-disable;
			output-low;
		};
	};
};

&blsp1_spi4 {
	pinctrl-0 = <&spi_3_pins &quartz_pins>;
	pinctrl-names = "default";
};
```

This should work while being mostly compliant to DT bindings